docs/specifications/tests/DNSSEC-TP/dnssec16.md

```diff
@@ -0,0 +1,340 @@
+# DNSSEC16: Validate CDS and CDNSKEY
+
+## Test case identifier
+**DNSSEC16**
+
+## Objective
+
+CDS and CDNSKEY record types are defined in [RFC 7344] and [RFC 8878].
+Both record types are optional in a zone. The objective of this test
+case is to verify that they are they are valid. This Test case is
+only relevant if the zone has either CDS or CDNSKEY record or both.
+
+If an CDS record is included in the zone, the corresponding CDNSKEY
+record should also be included ([RFC 7344][RFC 7344, section 4],
+section 4).
+
+The CNS and CDNSKEY RRsets should be consistent between all name
+servers for the zone in question.
+
+If there are both CDS RRs and CDNSKEY RRs in the zone they must match
+in content ([RFC 7344][RFC 7344, section 4], section 4). It means that
+both must be derived from the same DNSKEY or both being "delete" CDS
+and CDNSKEY.
+
+If a name server has issues covered by [Basic04] (basic name server
+issues) no messages will be outputted from this test case.
+
+It is assumed that the Child Zone has been tested by [DNSSEC15] and
+that the servers give the same responses.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+* If no CDS or CDNSKEY is found, the test case is nor run.
+* [ERROR] message if there is no DNSKEY RRset.
+* [WARNING] message if a CDS record does not match any
+  DNSKEY in DNSKEY RRset (except for "delete" CDS).
+* [WARNING] message if a CDNSKEY record does not match any
+  DNSKEY in DNSKEY RRset (except for "delete" CDNSKEY).
+* [ERROR] message if CDS RRset is not signed.
+* [ERROR] message if CDNSKEY RRset is not signed.
+* [ERROR] message if CDS RRset is signed with an invalid RRSIG.
+* [ERROR] message if CDNSKEY RRset signed with an invalid RRSIG.
+* [ERROR] message if CDS RRset is signed but not by a key in DNSKEY
+  RRset.
+* [ERROR] message if CDNSKEY RRset is signed but not by a key in DNSKEY
+  RRset.
+* [WARNING] message if CDS RRset is signed by a key in the DNSKEY RRset
+  but not with one that has signed the DNSKEY RRset.
+* [WARNING] message if CDNSKEY RRset is signed by a key in the DNSKEY
+  RRset but not with one that has signed the DNSKEY RRset.
```

Comment:
Actually RFC7344 says "MUST be signed with a key that is represented in both the current DNSKEY and DS RRsets,"

But that only applies if a DS set exists.

Reply:
And that could be the case here.